### PR TITLE
akonhilas/APPEALS-34880

### PR DESCRIPTION
**Pull request by konhilas-ariana**
## Resolves Webex Services Fails to Handles Both Virtual Hearings and Hearing Days
https://jira.devops.va.gov/browse/APPEALS-34880

# Description
Daily docket page presents a 500 error. This is due to a failure that occurs whenever attempting to generate Webex conference links for the hearing day.

## Acceptance Criteria
- [ ] Code compiles correctly

## Testing Plan
1. Go to https://jira.devops.va.gov/browse/APPEALS-34885

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Backend

## Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
  * [ ] Check that calls are wrapped in MetricService record block
* [ ] Check that all configuration is coming from ENV variables
  * [ ] Listed all new ENV variables in description
  * [ ] Worked with or notified System Team that new ENV variables need to be set
* [ ] Update Fakes
* [ ] For feature branches: Was this tested in Caseflow UAT

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [ ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

File: app/services/external_api/webex_service.rb [hunk 1]
```diff
@@ -16,8 +16,8 @@ def create_conference(virtual_hearing)
     body = {
       "jwt": {
         "sub": virtual_hearing.subject_for_conference,
-        "Nbf": virtual_hearing.hearing.scheduled_for.beginning_of_day.to_i,
-        "Exp": virtual_hearing.hearing.scheduled_for.end_of_day.to_i
+        "Nbf": virtual_hearing.nbf,
+        "Exp": virtual_hearing.exp
```
Comment (ThorntonMatthew) [inline app/services/external_api/webex_service.rb:20]: I don't think the keys are case-sensitive, but I'm hesitant to take any chances.

```suggestion
        "nbf": virtual_hearing.nbf,
        "exp": virtual_hearing.exp
```

Comment (konhilas-ariana) [reply app/services/external_api/webex_service.rb:20]: I know I'm scared to touch the keys but I will change it if you want me to!

Comment (ThorntonMatthew) [reply app/services/external_api/webex_service.rb:20]: I'll test it on my GFE.

Comment (ThorntonMatthew) [reply app/services/external_api/webex_service.rb:20]: The capital letters didn't make a difference. You can disregard the original comment.

**Review by ThorntonMatthew — APPROVED** [review]
THANK YOU!!

**Pull request merged**